xarray/core/dataset.py

```diff
@@ -3024,11 +3005,11 @@ def expand_dims(
                 # it will be promoted to a 1D coordinate with a single value.
                 variables[k] = v.set_dims(k).to_index_variable()
 
-        new_dims = self._dims.copy()
-        new_dims.update(dim)
-
-        return self._replace_vars_and_dims(
-            variables, dims=new_dims, coord_names=coord_names
+        return self._replace_with_new_dims(
+            # TODO: fix indexes, I think by passing nothing (i.e. default)
+            variables=variables,
+            coord_names=coord_names,
+            indexes=None,
```

Comment:
I think the right way to set `indexes` here would be to copy them from the original dataset and then add/modify indexes as needed for any expanded dimensions.